tests/python/gaia-ui-tests/gaiatest/apps/wallpaper/app.py

```diff
@@ -17,7 +17,6 @@ def tap_wallpaper_by_index(self, index):
         Wait(self.marionette).until(
             lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= (index + 1),
             message='%d wallpaper(s) not present after timeout' % (index + 1))
-        self.tap_element_from_system_app(
-            self.marionette.find_elements(*self._stock_wallpapers_locator)[index])
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != self.name)
+        self.marionette.find_elements(*self._stock_wallpapers_locator)[index].tap()
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != self.manifest_url)
```

Comment:
Use self.wait_to_be_displayed() here. But the wallpaper app needs a manifest_url property.